Fix/LP-161 phone number components

JoseGR98 opened the pull request:
LP-161
https://devaptivist.atlassian.net/browse/LP-161
Phone number component reworked. The error on previous version is not showing anymore. However, there is still another error, even when the keyboard is "numbers only", if user adds text it can break. However, this will be updated and checked on the ViewModel (suggested by Sergio).
Files modified:

TextField.kt --> fun LPPhoneNumberText()
Strings.kt --> 1 string added
PhoneNumberFormat.kt --> Changes to almost the entire file
RepairView --> fun LPPhoneNumberText()
Pictures of component, different states:

![Foto1](https://user-images.githubusercontent.com/25538206/233694020-9e6c486a-56c0-4876-8a6e-396681b9a203.png)
![Foto2](https://user-images.githubusercontent.com/25538206/233694022-22dc0ee9-042e-41e2-8cbb-9a106028eeae.png)
![Foto3](https://user-images.githubusercontent.com/25538206/233694023-6bdfb464-d7cf-4225-868a-93fab664e458.png)
![Foto4](https://user-images.githubusercontent.com/25538206/233694026-f577039b-fd4d-4cc8-82be-118451b5d6af.png)

Commit:
RepairViewModel.kt added. Progress on ViewModel task

Commit:
Merge branch 'develop' into feature/lp-114-repair-request-viewmodel

Commit:
Merge branch 'develop' into feature/lp-114-repair-request-viewmodel

Commit:
Merge branch 'develop' into fix/lp-161-phone-number-components

Commit:
Merge branch 'develop' into fix/lp-161-phone-number-components

Commit:
Merge branch 'develop' into fix/lp-161-phone-number-components

Commit:
Merge branch 'develop' into fix/lp-161-phone-number-components

Commit:
LP-161 Bug Fix progress

Commit:
LP-161 Bug Fix progress

carlosph816 commented on feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt:
```diff
@@ -63,24 +66,27 @@ fun RepairView() {
-                }
+                },
+                isValid = isAValidNumber,
+                supportTextError = "Not a valid number"
```
Add this text to string resources

JoseGR98 replied:
Done! :)

Commit:
Merge branch 'develop' into fix/lp-161-phone-number-components

Commit:
LP-161 Bug Fix progress
TextField.kt - reworked field
strings.xml - one string added
PhoneNumberFormat.kt - to have phone number format
RepairView.kt - modified to recognize new parameters

melissaquezada94 commented on core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:
```diff
@@ -304,49 +302,54 @@ fun DropdownTextField(
             }
             onPhoneChange(it)
         },
+        isError = isValid,
```
If this `isValid` is false means it has a error?

JoseGR98 replied:
My apologies. It should be "isNotValid", I will re-name it to avoid confusion. The new change:
If "isNotValid" is true, will mean that the values is not valid.
If "isNotValid" is false, will mean that the value is valid.

JoseGR98 replied:
It has been changed :)

melissaquezada94 commented on core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:
```diff
@@ -304,49 +302,54 @@ fun DropdownTextField(
         },
+        isError = isValid,
+        supportingText = {
+            if (isValid)
```
Are you showing an error if the number is not valid?

JoseGR98 replied:
Yes, the error is the red text that shows on the 3rd image. I will just rename the variable to avoid confusions

JoseGR98 replied:
It has been changed :)

melissaquezada94 commented on feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt:
```diff
@@ -29,6 +29,9 @@ fun RepairView() {
     var isAValidDescription by remember {
         mutableStateOf(false)
     }
+    var isAValidNumber by remember {
```
You can declare the variable like `isValidPhone`

JoseGR98 replied:
Ok, will change it

JoseGR98 replied:
It has been changed :)

melissaquezada94 commented on feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt:
```diff
@@ -0,0 +1,36 @@
+package com.iteneum.repair.data
```
The viewModel is going to be created in the next spring so I think you don't need this class

JoseGR98 replied:
I added it, by mistake. Will remove it

JoseGR98 replied:
It has been changed :)

JoseGR98 commented:
Already uploaded corrections. New PR

carlosph816 commented on core/designsystem/src/main/java/com/iteneum/designsystem/components/phonenumbertext/PhoneNumberFormat.kt:
```diff
@@ -1,44 +1,65 @@
+
+        val output = buildString {
+            for ((index, char) in text.withIndex()) {
+                Log.d("ATTENTION", "TEXT IS -> ${text.text}")
```
Let's delete the log

JoseGR98 replied:
Done! :)

carlosph816 commented on core/designsystem/src/main/java/com/iteneum/designsystem/components/phonenumbertext/PhoneNumberFormat.kt:
```diff
@@ -1,44 +1,65 @@
+                Log.d("ATTENTION", "TEXT IS -> ${text.text}")
+                if (text.text.length <= 10 && text.text.isDigitsOnly()) {
+                    append(char)
+                    Log.d("CHECK_NUMBER", "APPENDABLE")
```
Here too

JoseGR98 replied:
Done! :)

carlosph816 commented on core/designsystem/src/main/java/com/iteneum/designsystem/components/phonenumbertext/PhoneNumberFormat.kt:
```diff
@@ -1,44 +1,65 @@
+                    append(char)
+                    Log.d("CHECK_NUMBER", "APPENDABLE")
+                } else {
+                    Log.d("ERROR", "NOT APPENDABLE")
```
Here too

JoseGR98 replied:
Done! :)

carlosph816 commented on core/designsystem/src/main/java/com/iteneum/designsystem/components/phonenumbertext/PhoneNumberFormat.kt:
```diff
@@ -1,44 +1,65 @@
+                }
+                if (index == separatorPositionA || index == separatorPositionB) {
+                    append(separator)
+                    Log.d("CHECK_SEPARATOR", "APPENDABLE")
```
Here too

JoseGR98 replied:
Done! :)

Commit:
Merge branch 'develop' into fix/lp-161-phone-number-components

carlosph816 commented on core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:
```diff
@@ -304,49 +304,54 @@ fun DropdownTextField(
-        visualTransformation = { mobileNumberFilter(it) },
+        visualTransformation = {
+            phoneNumberTransformation.filter(AnnotatedString(phoneNumberText))
+                               },
```
Reformat code please

JoseGR98 replied:
Done! :)

Commit:
LP-161 Bug Fix code reviewers comments update
TextField rename variables
RepairView rename variables
PhoneNumberFormat corrections
RepairViewModel deleted

melissaquezada94 reviewed (approved):
Well done!

Commit:
Merge branch 'develop' into fix/lp-161-phone-number-components

Commit:
LP-161 Bug Fix code reviewers comments update 2
String added. Errors solved.

Commit:
LP-161 Bug Fix code reviewers comments update 3.
Reformat code.
